R/zchunk_LB165.ag_water_R_C_Y_GLU_irr.R

```diff
@@ -0,0 +1,407 @@
+    iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
+    FAO_ag_items_PRODSTAT <- get_data(all_data, "aglu/FAO_ag_items_PRODSTAT")
+    L100.Water_footprint_m3 <- get_data(all_data, "L100.Water_footprint_m3")
+    Mekonnen_Hoekstra_Rep47_A2 <- get_data(all_data, "aglu/Mekonnen_Hoekstra_Rep47_A2")
+    Rohwer_2007_IrrigationEff <- get_data(all_data, "aglu/Rohwer_2007_IrrigationEff")
+    L100.LDS_ag_prod_t <- get_data(all_data, "L100.LDS_ag_prod_t")
+    L151.ag_irrProd_t_ctry_crop <- get_data(all_data, "L151.ag_irrProd_t_ctry_crop")
+    L151.ag_rfdProd_t_ctry_crop <- get_data(all_data, "L151.ag_rfdProd_t_ctry_crop")
+    L151.ag_irrHA_ha_ctry_crop <- get_data(all_data, "L151.ag_irrHA_ha_ctry_crop")
+
+
+    # Perform computations
+
+    # The method here is simple in principle: we are taking inventory
+    # estimates of either water demand coefficients by country and crop, or aggregated
+    # gridded volumes of water use by country, GLU, and crop, and using these estimates to
+    # calculate average water consumption coefficients by GCAM region, crop, GLU, and
+    # irrigation level.
+    #
+    # All data represent a statistical year around 2000, and the GTAP/Monfreda data were
+    # used in constructing the gridded and national inventory data, so this is the appropriate
+    # production data to use.
+    #
+    # The major methodological decisions relate to how to estimate the coefficients by GLU for
+    # crops that are not available in the gridded inventory, how to extrapolate from the given
+    # crops to the fodder crops, and how to assign priority for the crops included in both the
+    # gridded and national inventories. These decisions are documented below.
+
+    # The inventory data disaggregate green, blue, and gray water. Gray water indicates
+    # downstream pollution; this is not considered in GCAM and has not been included in the
+    # gridded data processing. Blue water is assigned to only irrigated production, but green
+    # water applies to both irrigated and rainfed production, as the two management technologies
+    # are not disaggregated in the inventory data. For rainfed crops, total biophysical = green,
+    # and for irrigated crops, total biophysical = blue + green.
+
+    # Initial data cleaning (original lines 55-57)
+    L100.Water_footprint_m3 %>%
+      rename(MH_crop = GTAP_crop) %>%
+      left_join_error_no_match(select(FAO_ag_items_PRODSTAT, MH_crop, GTAP_crop), by = "MH_crop") ->
+      L165.Water_footprint_m3
+
+    # Re-cast nation-level data so that blue and green are represented as data columns (original lines 59-64)
+    Mekonnen_Hoekstra_Rep47_A2 %>%
+      gather(iso, coef_m3t, -FAO_crop, -water_type) %>%
+      mutate(coef_m3kg = coef_m3t / CONV_T_KG) %>%
+      select(iso, FAO_crop, water_type, coef_m3kg) %>%
+      # Set all missing values to zero -- this may be re-visited at another point
+      # We may want to substitute default values from the MH2011 table
```

Comment:
I'd recommend just leaving it as a comment for now, and providing some more information. Hard to see this really rising to a priority level but here's the info in case:
The source data table has 29,725 observations (country x crop), of which 24,063 are missing values, and of these missing values, 321 did actually have crop production in Monfreda's inventory, 13 of which had > 1 Mt of production. We are using this dataset to supplement a global gridded inventory with 18 major crops. Filtering those out, there are still 4 observations with between 1 and 3 Mt of production, all in Nigeria: CitrusFrtNES, CowPeasDry, Plantains, and FrtFrshNES.
If we want to replace these with defaults, the blue water coef (i.e., the important one) will still be zero, because it's in Nigeria, where ~100% of crop production is rainfed (according to the inventories). So, if we do implement a different method, we'll get a slightly higher estimate of biophysical water consumption for MiscCrop in Africa_Western.